### PR TITLE
HDFS-16640. RBF: Show datanode IP list when click DN histogram in Router

### PR DESCRIPTION
**Pull request by wzhallright**
### Description of PR
I'm not sure if need to show a list of ip's when click dn histogram. I just saw the open_hostip_list() was called, but not implemented. So I ported the code from NN UI.
JIRA: https://issues.apache.org/jira/browse/HDFS-16640

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 50s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  jshint  |   0m  1s |  |  jshint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  63m 25s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  83m  3s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shadedclient  |  19m  7s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 52s |  |  The patch does not generate ASF License warnings.  |
|  |   | 106m 36s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4488/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4488 |
| Optional Tests | dupname asflicense shadedclient codespell detsecrets jshint |
| uname | Linux f5fd1e8ef206 4.15.0-169-generic #177-Ubuntu SMP Thu Feb 3 10:50:38 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 73c08effebd6823ddbb21dbb722f6059c4e1a36e |
| Max. process+thread count | 757 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-rbf U: hadoop-hdfs-project/hadoop-hdfs-rbf |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4488/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/webapps/router/federationhealth.js [hunk 1]
```diff
@@ -552,3 +552,45 @@
     load_page();
   });
 })();
+
+function open_hostip_list(x0, x1) {
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/webapps/router/federationhealth.js]: Where is this called?

Comment (ayushtkn) [reply hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/webapps/router/federationhealth.js]: guess here:
https://github.com/wzhallright/hadoop/blob/73c08effebd6823ddbb21dbb722f6059c4e1a36e/hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/webapps/router/federationhealth.js#L376

**Review by ayushtkn — APPROVED** [review]
Code seems exact copy from dfshealth.js, that is what we are doing for router UI, seems correct to me.
Good if you can attach a screenshot as well

**Comment by goiri** [comment]
> Code seems exact copy from dfshealth.js, that is what we are doing for router UI, seems correct to me. Good if you can attach a screenshot as well

Not an expert on javascript but it would be good to extract some if this common code.

**Comment by wzhallright** [comment]
> 
@ayushtkn 
Of course, I uploaded my test screenshot
image

**Comment by ayushtkn** [comment]
>it would be good to extract some if this common code.

Not an expert with Javascript either, but I feel this should be possible. If we extract all these common methods into some other Javascript file say Utils.js and then add that in both dfsHealth.html and FederationHealth.html before these .js files like:

```
</script><script type="text/javascript" src="/static/util.js">
```

then may be we can get away from this copy-paste stuff

@wzhallright can you give it a try for the methods that you copied or if you know a better way to avoid this duplication....

**Comment by wzhallright** [comment]
> 

I'm not good at JS... But I can try to modify

**Comment by wzhallright** [comment]
@ayushtkn Add histogram-hostip.js, after modified the screenshots. Please take a review, Thanks!
RBF.  
image.  
NN
image

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 46s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  jshint  |   0m  0s |  |  jshint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 21s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  27m  0s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  62m  4s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  3s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shadedclient  |  19m 21s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 51s |  |  The patch does not generate ASF License warnings.  |
|  |   |  87m 35s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4488/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4488 |
| Optional Tests | dupname asflicense shadedclient codespell detsecrets xmllint jshint |
| uname | Linux bbd2d63a1905 4.15.0-169-generic #177-Ubuntu SMP Thu Feb 3 10:50:38 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 550cf7145ea36948de7ada1e48f3c534d1760850 |
| Max. process+thread count | 558 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs hadoop-hdfs-project/hadoop-hdfs-rbf U: hadoop-hdfs-project |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4488/2/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 52s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  jshint  |   0m  0s |  |  jshint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m  6s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  28m 16s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  66m  3s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  2s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shadedclient  |  22m  1s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   |  93m 53s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4488/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4488 |
| Optional Tests | dupname asflicense shadedclient codespell detsecrets xmllint jshint |
| uname | Linux f4567394809c 4.15.0-175-generic #184-Ubuntu SMP Thu Mar 24 17:48:36 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 550cf7145ea36948de7ada1e48f3c534d1760850 |
| Max. process+thread count | 578 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs hadoop-hdfs-project/hadoop-hdfs-rbf U: hadoop-hdfs-project |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4488/3/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by goiri — APPROVED** [review]

**Review by ayushtkn — APPROVED** [review]
LGTM

**Pull request merged**